tawazi/_dag/dag.py

```diff
@@ -529,13 +482,22 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> RVDAG:
         """
         if kwargs:
             raise TawaziUsageError(f"currently DAG does not support keyword arguments: {kwargs}")
-        # 1. generate the subgraph to be executed
-        graph = self.make_subgraph()
 
-        # 2. copy the ExecNodes
-        call_xn_dict = self.make_call_xn_dict(*args)
+        graph = deepcopy(self.graph_ids)
+
+        # add debug nodes
+        if cfg.RUN_DEBUG_NODES:
+            debug_ids = self.graph_ids.include_debug_nodes(graph.leaf_nodes)
+            graph = self.graph_ids.subgraph(list(graph.nodes) + debug_ids).copy()
```

Comment:
we should be consistent. we currently mix node and id names in the code. Do you have any idea how to do it ?